strapi/strapi
# [Upgrade-Tool] Refactor the CLI and add codemods commands

### What does it do?

- Clean-up & reorganize the upgrade-tool CLI
- Refactors the codemods command, split it in two and add sub-commands
	- `codemods run [uid]` to run codemods (optionally provide a single codemod UID)
	- `codemods ls` to get a list of available codemods
- other small refactoring in codemods related modules (`codemod-runner`, `codemod-repository,` `codemod`)

### Why is it needed?

Allow listing available codemods and running specific ones, especially useful when developing codemods for a version that's not available now.

### How to test it?

- Try playing with the commands (`codemods ls`, `codemods run [UID]`) and make sure everything works according to the help description
